pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -652,6 +640,8 @@ func getBackupRestoreEnvVars(val Values) []corev1.EnvVar {
 		return env
 	}
 
+	env = append(env, getEnvVarFromValue("STORAGE_CONTAINER", pointer.StringDeref(val.BackupStore.Container, "")))
```

Comment:
Don't we need the check here for `val.BackupStore != nil` to avoid pointer dereference error?

Reply:
@shreyas-s-rao There is a nil check
```go
if storeValues == nil {
		return env
}
```
If the storeValues which is `val.BackupStore` is nil then you there is an early return from this function. So its fine not to check for nil again. Also `pointer.StringDeref` anyways take care of the case where `Container` could also be nil.

Reply:
Oops, I missed that, thanks.